packages/core/test/parser/util.spec.ts

```diff
@@ -43,54 +44,92 @@ function literal(
-			parsers: [literal('foo'), literal('bar')],
-			content: 'bar',
-			parserToString: 'foo | bar',
-		},
-		{
-			parsers: [literal('foo'), literal('bar')],
-			content: 'qux',
-			parserToString: 'foo | bar',
-		},
-		{
-			parsers: [literal('foo', 'correct', 1), literal('foo', 'wrong', 1)],
-			content: 'foo',
-			parserToString: 'foo*1 | foo*1',
-		},
-		{
-			parsers: [literal('foo', 'correct', 1), literal('foo', 'wrong', 2)],
-			content: 'foo',
-			parserToString: 'foo*1 | foo*2',
-		},
-		{
-			parsers: [literal('foo', 'wrong', 2), literal('foo', 'correct', 1)],
-			content: 'foo',
-			parserToString: 'foo*2 | foo*1',
-		},
-	]
-	for (const { content, parsers, parserToString } of suites) {
-		it(
-			`Parse "${
-				showWhitespaceGlyph(
-					content,
-				)
-			}" with "${parserToString}"`,
-			() => {
-				const parser = any(parsers)
-				snapshot(testParser(parser, content))
+describe('core util parsers', () => {
+	describe('any()', () => {
+		const suites: {
+			content: string
+			parsers: [Parser<AstNode>, ...Parser<AstNode>[]]
+			parserToString: string
+		}[] = [
+			{
+				parsers: [literal('foo'), literal('bar')],
+				content: 'foo',
+				parserToString: 'foo | bar',
 			},
-		)
-	}
+			{
+				parsers: [literal('foo'), literal('bar')],
+				content: 'bar',
+				parserToString: 'foo | bar',
+			},
+			{
+				parsers: [literal('foo'), literal('bar')],
+				content: 'qux',
+				parserToString: 'foo | bar',
+			},
+			{
+				parsers: [literal('foo', 'correct', 1), literal('foo', 'wrong', 1)],
+				content: 'foo',
+				parserToString: 'foo*1 | foo*1',
+			},
+			{
+				parsers: [literal('foo', 'correct', 1), literal('foo', 'wrong', 2)],
+				content: 'foo',
+				parserToString: 'foo*1 | foo*2',
+			},
+			{
+				parsers: [literal('foo', 'wrong', 2), literal('foo', 'correct', 1)],
+				content: 'foo',
+				parserToString: 'foo*2 | foo*1',
+			},
+		]
+		for (const { content, parsers, parserToString } of suites) {
+			it(
+				`Parse "${
+					showWhitespaceGlyph(
+						content,
+					)
+				}" with "${parserToString}"`,
+				() => {
+					const parser = any(parsers)
+					snapshot(testParser(parser, content))
+				},
+			)
+		}
+	})
+
+	describe('dumpErrors()', () => {
+		const suites: {
+			name: string
+			parser: Parser<AstNode>
+			content: string
+			expectedErrors: any[]
+		}[] = [
+			{
+				name: 'should output errors when not wrapped with `dumpErrors()`',
+				parser: boolean,
+				content: 'bar',
+				expectedErrors: [{
+					range: {
+						start: 0,
+						end: 0,
+					},
+					message: 'Expected “false” or “true”',
+					severity: 3,
+				}],
+			},
+			{
+				name: 'should not output errors when wrapped with `dumpErrors()`',
+				parser: dumpErrors(boolean),
+				content: 'bar',
+				expectedErrors: [],
+			},
+		]
+		for (const { name, content, parser, expectedErrors } of suites) {
+			it(name, () => {
+				const res = testParser(parser, content)
+				snapshot(res)
+				assert.deepEqual(res.errors, expectedErrors)
```

Comment:
If you're snapshotting the result, there is no reason to check for the expectedErrors again.

Reply:
Additionally, this file is already called `core/test/parser/util`, so it's unnecessary to wrap it in `describe('core util parsers'`

Reply:
> If you're snapshotting the result, there is no reason to check for the expectedErrors again.

i kinda like the redundancy in the check -- snapshots are definitely nice to get a big picture of what the parser outputs! but i like also checking for the main thing we want the parser to be doing (e.g. do stuff around errors in this case)

i'm pretty sure we actually have other tests that intermingle `snapshot` and `assert`, so this wouldnt be the first time we do that redundantly:

https://github.com/SpyglassMC/Spyglass/blob/b66297d1f6c6a7bfa34ad2983f55a5e1cba8a278/packages/core/test/common/StateProxy.spec.ts#L88-L110

Reply:
I was mostly basing this on the above `any()` tests in the same file. I'm not sure how the mentioned test supports your argument. It snapshots the `proxy` object, but then asserts the `testObj` and `barSymbol`. The `expectedErrors` you wrote in the test are exactly the same as the ones in the snapshot, so they're useless.